org.mwc.cmap.legacy/src/MWC/GenericData/TimePeriod.java

```diff
@@ -171,6 +173,31 @@ public final HiResDate getStartDTG()
       return _startDTG;
     }
 
+    @Override
+    public boolean equals(Object obj)
+    {
+      if (!(obj instanceof TimePeriod))
+        return false;
+      if (obj == this)
+        return true;
+      TimePeriod tp = (TimePeriod) obj;
+      if(tp.getStartDTG().equals(this.getStartDTG()) && tp.getEndDTG().equals(this.getEndDTG()))
+        return true;
+      else
+        return false;
+    }
+
+
+    @Override

```

Comment:
@theanuradha - I'm not very experienced at *hash codes*.  Does this `hashCode()` function look correct/valid?

p.s. I don't know how/why examples seem to introduce values like **7** and **31** - but I've used them anyway,